### PR TITLE
fix: typos, naming clashes, error messages and deprecations

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Also GoLang code deserves love ....... 

## Motivation and Context
:love_hotel: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: ocis-pkg/log/log.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ type Logger struct {
 	zerolog.Logger
 }
 
-// LoggerFromConfig initializes a service-specific logger instance.
+// Configure initializes a service-specific logger instance.
```
Comment (rhafer) [inline ocis-pkg/log/log.go:53]: This is wrong the method is called `LoggerFromConfig`, so the comment should start with that.

**Review by rhafer — CHANGES_REQUESTED** [review]

File: services/proxy/pkg/config/config.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ type AuthMiddleware struct {
 	CredentialsByUserAgent map[string]string `yaml:"credentials_by_user_agent"`
 }
 
-// PoliciesMiddleware configures the proxy policies middleware.
+// PoliciesMiddleware configures the proxy policies' middleware.
```
[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply services/proxy/pkg/config/config.go:93]: it is the middleware of the proxy policies - the policies' middleware .... :shrug:

[the remark this answers is not in the document]
Comment (kobergj) [reply services/proxy/pkg/config/config.go:93]: Yes but what does the `'` mean?

[the remark this answers is not in the document]
Comment (kobergj) [reply services/proxy/pkg/config/config.go:93]: Ah now I get it. But that is not true.. It is the `policies middleware` of the proxy service

[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply services/proxy/pkg/config/config.go:93]:
```go
// PoliciesMiddleware configures the proxy's policies middleware.
```

**Review by kobergj — APPROVED** [review]

File: ocis-pkg/log/log.go [hunk 3]
```diff
@@ -123,27 +110,27 @@ func NewLogger(opts ...Option) Logger {
 			print(fmt.Sprintf("file could not be opened for writing: %s. error: %v", options.File, err))
 			os.Exit(1)
 		}
-		logger = logger.Output(f)
+		l = l.Output(f)
```
[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply ocis-pkg/log/log.go:113]: Quite common in the go code I have seen.....
But happy to apply a different naming standard if there is one.

Main issue here is that var names should not collide with imports.

[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply ocis-pkg/log/log.go:113]: But I should review this line anyhow...

[the remark this answers is not in the document]
Comment (kobergj) [reply ocis-pkg/log/log.go:113]: Makes sense here imho. The function is called `NewLogger` an `l` is the actual logger.

[the remark this answers is not in the document]
Comment (micbar) [reply ocis-pkg/log/log.go:113]: We agreed as a team that we do not want to have single chat var names. We should use speaking variable names. Exceptions are possible when a variable is only used in a for loop or so, when the readability of the code allows it.

[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply ocis-pkg/log/log.go:113]: This then?
```
zeroLogger = zeroLogger.Output(f)

[the remark this answers is not in the document]
Comment (kobergj) [reply ocis-pkg/log/log.go:113]: > when the readability of the code allows it.

Looks to me like this is the case here :wink:

[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply ocis-pkg/log/log.go:113]: I see no reason to diverge from go's own proposals. In the past we had so many issues in not following commonly accepted rules (e.g. in PHP where we did not follow PSR code style - contributions have been difficult for external developers).

I would just follow https://go.dev/wiki/CodeReviewComments#variable-names

Furthermore there are so many code blocks where variable names are nuts .... :shrug:

[the remark this answers is not in the document]
Comment (micbar) [reply ocis-pkg/log/log.go:113]: Yes, we have some bad namings where we have `resp1` or `resp2` or something like `n` for node all over a whole page.

@dragotin Has a very strong opinion on variable names. We should always choose readability over trying to make the name even shorter.

For example pointer receivers in go are always single letters, this is agreed best practise.

We are linking our coding guidelines in the Readme 😀 

https://github.com/owncloud/ocis/blob/master/CONTRIBUTING.md#styleguides

Specially this one https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html#_a_variables_name_should_describe_its_contents

**Review by rhafer — APPROVED** [review]
In general I am not a huge fan of these "let's fix everything in a single commit" PRs. They tend to make life harder when back-/forward porting stuff.

For the future I think it would be better to create separate commits to address specific problems. (Even it that would be a bit more work)

File: ocis-pkg/log/log.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply ocis-pkg/log/log.go:113]: > Specially this one https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html#_a_variables_name_should_describe_its_contents

following this `l` is just fine in the discussed code location
```


Naming the *Config parameter config is redundant. We know its a *Config, it says so right there.

In this case consider conf or maybe c will do if the lifetime of the variable is short enough.

**Pull request merged**